### PR TITLE
poetry dependencies for torch in macOS with M-series

### PR DESCRIPTION
**Pull request by Bhuvanesh-Verma**
This PR adds dependencies for torch in macOS with M-series. Currently tested on M1 only.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -36,9 +36,11 @@ pyarrow = "^13"
 pie-modules = ">=0.11.1,<0.13.0"
 torch = [
     # default
-    {version = "^2.1.0+cpu", source = "pytorch", markers = "sys_platform != 'darwin' or platform_machine != 'x86_64'"},
+    {version = "^2.1.0+cpu", source = "pytorch", markers = "sys_platform != 'darwin' or platform_machine != 'arm64' and platform_machine != 'x86_64'"},
```
Comment (ArneBinder) [inline pyproject.toml:39]: what is the precedence fop the `and` and `or` operators? having both next to each other without any brackets looks quite ambiguous...

**Review by ArneBinder — CHANGES_REQUESTED** [review]
please have a look

File: pyproject.toml (hunk 1 above)
Comment (Bhuvanesh-Verma) [reply pyproject.toml:39]: I'll add brackets for the `and` operation.

**Review by kai-car — COMMENTED** [review]
I am using **macOS x86_64** (Intel Chip) and for me the installation of the dependencies with `poetry install` still seems to work fine.

**Review by ArneBinder — APPROVED** [review]

**Pull request merged**